src/actions/attack.cpp

```diff
@@ -1606,15 +1606,15 @@ bool leadership_affects_self(const std::string& ability,const unit_map& units, c
 	unit_ability_list abil = un->get_abilities(ability, weapon, opp_weapon);
 	for(unit_ability_list::iterator i = abil.begin(); i != abil.end();) {
 		const std::string& apply_to = (*i->first)["apply_to"];
-		if(apply_to.empty() || apply_to == "both" || apply_to == "self") {
+		if(apply_to.empty() || apply_to == "both" || apply_to == "self" || (attacker && apply_to == "attacker") || (!attacker && apply_to == "defender")) {
```

Comment:
I think this would be easier to read compared to putting everything on one line:

```suggestion
		if(apply_to.empty() || apply_to == "both" || apply_to == "self") {
			return true;
		}
		if(attacker && apply_to == "attacker") {
			return true;
		}
		if(!attacker && apply_to == "defender") {
```